src/Language/PureScript/CoreImp/Optimizer/MagicDo.hs

```diff
@@ -54,6 +54,14 @@ magicDo'' effectModule C.EffectDictionaries{..} = everywhereTopDown convert
   -- Desugar whileE
   convert (App _ (App _ (App s1 f [arg1]) [arg2]) []) | isEffFunc C.whileE f =
     App s1 (Function s1 Nothing [] (Block s1 [ While s1 (App s1 arg1 []) (Block s1 [ App s1 arg2 [] ]), Return s1 $ ObjectLiteral s1 []])) []
+  -- Desugar forE
+  convert (App _ (App _ (App _ (App s1 f [loArg]) [hiArg]) [intToEff]) []) | isEffFunc C.forE f =
+    App s1 (Function s1 Nothing [] (Block s1 [ VariableIntroduction s1 lo (Just loArg), VariableIntroduction s1 hi (Just hiArg), VariableIntroduction s1 fn (Just intToEff), For s1 counter (Var s1 lo) (Var s1 hi) (Block s1 [App s1 (App s1 (Var s1 fn) [Var s1 counter]) []]), Return s1 $ ObjectLiteral s1 []])) []
```

Comment:
Would it hurt to split this up onto multiple lines? 😜

Reply:
It would not :). Will do so.